src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 1) {
+      return String(this);
+    }
+
+    let sep = separator;
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    if (this[0] === null || this[0] === undefined) {
+      this[0] = '';
+    }
```

Comment:
you repeat the same logic inside the loop below, can we make it only in one place?